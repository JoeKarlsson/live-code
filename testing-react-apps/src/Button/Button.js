/*
  eslint
	react/jsx-no-bind: 0
*/

import React from 'react';
import PropTypes from 'prop-types';
import multiply from '../multiply';

const Button = (props) => {
	const { alertNumber } = props;

	return (
		<button
			className="newclassname"
			id="react-button"
			onClick={alertNumber.bind(this, multiply(3, 2))}
		>
			Multiply
			{props.data}
		</button>
	);
};

Button.propTypes = {
	alertNumber: PropTypes.func.isRequired,
	data: PropTypes.string.isRequired,
};

Button.defaultProps = {};

export default Button;
