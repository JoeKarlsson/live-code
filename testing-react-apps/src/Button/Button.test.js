/*
  eslint
	no-undef: 0
*/

import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Button from './Button';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
	describe('Snapshot test', () => {
		it('match the snapshot', () => {
			const mockAlertNumber = jest.fn();
			const component = renderer.create(<Button data="Joe Rules" alertNumber={mockAlertNumber} />);
			const tree = component.toJSON();
			expect(tree).toMatchSnapshot();
		});

		// it('match the snapshot', () => {
		// 	const mockAlertNumber = jest.fn();
		// 	const component = renderer.create(<Button data="hello wolrd" alertNumber={mockAlertNumber} />);
		// 	const tree = component.toJSON();
		// 	expect(tree).toMatchSnapshot();
		// });
	});

	describe('Integration test', () => {
		it('should run the multiply function on click', () => {
			const mockAlertNumber = jest.fn();
			const component = mount(<Button data="Joe Rules" alertNumber={mockAlertNumber} />);
			component.find('button').simulate('click');
			expect(mockAlertNumber).toHaveBeenCalledTimes(1);
		});
	});
});
