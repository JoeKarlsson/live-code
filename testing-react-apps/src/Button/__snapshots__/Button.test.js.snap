// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot test match the snapshot 1`] = `
<button
  className="newclassname"
  id="react-button"
  onClick={[Function]}
>
  Multiply
  Joe Rules
</button>
`;
